sdk/python/sawtooth_sdk/consensus/zmq_driver.py

```diff
@@ -133,9 +133,9 @@ def _register(self):
                 'Registration failed with status {}'.format(response.status))
 
     def _wait_until_active(self):
+        future = self._stream.receive()
         while True:
             try:
-                future = self._stream.receive()
                 message = future.result(1)
             except concurrent.futures.TimeoutError:
                 continue
```

Comment:
Please correct if I understood it wrong, is the issue related to future variable being overwritten by a new received message while it was still waiting for earlier future response to complete?

Reply:
That is correct